File: subprojects/frida-node/examples/rpc.ts
import frida from "frida";

const processName = process.argv[2];

const source = `
rpc.exports = {
  hello() {
    return 'Hello';
  },
  failPlease() {
    oops;
  }
};
`;

async function main() {
    const session = await frida.attach(processName);

    const script = await session.createScript(source);
    await script.load();

    try {
        const api = script.exports;

        console.log("[*] api.hello() =>", await api.hello());

        await api.failPlease();
    } finally {
        await script.unload();
    }
}

main()
    .catch(e => {
        console.error(e);
    });

File: subprojects/frida-python/examples/rpc.py
import frida

session = frida.attach("Twitter")
script = session.create_script("""\
rpc.exports = {
  hello: function () {
    return 'Hello';
  },
  failPlease: function () {
    oops;
  }
};
""")
script.load()
api = script.exports_sync
print("api.hello() =>", api.hello())
api.fail_please()
